scripts/distribute-rewards.js

```diff
@@ -17,19 +17,52 @@ const BATCHER_ABI = [
   }
 ]
 
-const marketAddress = "";
 const transactionBatcher = "0xA73A872eFD768bb23efb24CEeB9e330bcCA259D6";
 const subgraphUrl = "https://api.thegraph.com/subgraphs/name/prodeapp/prodeapp";
+const graph = new GraphQLClient(subgraphUrl);
 
 async function main() {
   const chainId = hre.network.config.chainId;
   const [signer] = await ethers.getSigners();
 
-  console.log("Account balance:", (await signer.getBalance()).toString());
+  console.log("Account balance:", ethers.utils.formatUnits((await signer.getBalance()).toString(), 18));
   console.log("Chain Id:", chainId);
 
-  const graph = new GraphQLClient(subgraphUrl);
+  // fetch candidate markets
+  let {markets} = await graph.request(
+    gql`
+      query marketsQuery {
+        markets(where: {resultSubmissionPeriodStart_gt: 0}) {
+          id
+          name
+          resultSubmissionPeriodStart
+          submissionTimeout
+          bets {
+            claim
+          }
+        }
+      }
+    `
+  );
+
+  const now = Math.floor(Date.now() / 1000);
+
+  markets = markets.filter(market => now > (Number(market.resultSubmissionPeriodStart) + Number(market.submissionTimeout)))
+
+  if (markets.length === 0) {
+    console.log('No markets found');
+    return;
+  }
 
+  for (const market of markets) {
+    console.log(`${market.name} (${market.id})`);
+    await processMarket(market.id, market.bets.filter(bet => bet.claim).length, signer);
+  }
+}
+
+async function processMarket(marketAddress, totalClaimed, signer) {
+  // TODO: we are reading all the markets, we will need to optimize this if we reach a big amount of markets
+  // (maybe keep track of the markets distributed or limit to the markets closed the last few days)
```

Comment:
Both seem good ideas. For now I'd go for the simplest one: 

`markets(where: {resultSubmissionPeriodStart_gt: Today - 2 weeks}) {`